tests/cnf/core/network/metallb/tests/bgp-remote-as-dynamic.go

```diff
@@ -0,0 +1,215 @@
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/metallb/internal/frr"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/metallb/internal/metallbenv"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/metallb/internal/tsparams"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var _ = Describe("BGP remote-dynamicAS", Ordered, Label(tsparams.LabelDynamicRemoteASTestCases),
+	ContinueOnFailure, func() {
+		var (
+			err           error
+			dynamicASiBGP = "internal"
+			dynamicASeBGP = "external"
+		)
+
+		BeforeAll(func() {
+			By("Getting MetalLb load balancer ip addresses")
+			ipv4metalLbIPList, ipv6metalLbIPList, err = metallbenv.GetMetalLbIPByIPStack()
+			Expect(err).ToNot(HaveOccurred(), tsparams.MlbAddressListError)
+
+			By("List CNF worker nodes in cluster")
+			cnfWorkerNodeList, err = nodes.List(APIClient,
+				metav1.ListOptions{LabelSelector: labels.Set(NetConfig.WorkerLabelMap).String()})
+			Expect(err).ToNot(HaveOccurred(), "Failed to discover worker nodes")
+
+			By("Selecting worker node for BGP tests")
+			workerLabelMap, workerNodeList = setWorkerNodeListAndLabelForBfdTests(cnfWorkerNodeList, metalLbTestsLabel)
+			ipv4NodeAddrList, err = nodes.ListExternalIPv4Networks(
+				APIClient, metav1.ListOptions{LabelSelector: labels.Set(workerLabelMap).String()})
+			Expect(err).ToNot(HaveOccurred(), "Failed to collect external nodes ip addresses")
+
+			err = metallbenv.IsEnvVarMetalLbIPinNodeExtNetRange(ipv4NodeAddrList, ipv4metalLbIPList, nil)
+			Expect(err).ToNot(HaveOccurred(), "Failed to validate metalLb exported ip address")
+
+			By("Listing master nodes")
+			masterNodeList, err = nodes.List(APIClient,
+				metav1.ListOptions{LabelSelector: labels.Set(NetConfig.ControlPlaneLabelMap).String()})
+			Expect(err).ToNot(HaveOccurred(), "Fail to list master nodes")
+			Expect(len(masterNodeList)).To(BeNumerically(">", 0),
+				"Failed to detect master nodes")
+		})
+
+		AfterAll(func() {
+
+			By("Remove custom metallb test label from nodes")
+			removeNodeLabel(workerNodeList, metalLbTestsLabel)
+		})
+
+		Context("single hop", func() {
+
+			var (
+				addressPool                  []string
+				hubIPv4ExternalAddresses     = []string{"172.16.0.10", "172.16.0.11"}
+				externalAdvertisedIPv4Routes = []string{"192.168.100.0/24", "192.168.200.0/24"}
+				externalAdvertisedIPv6Routes = []string{"2001:100::0/64", "2001:200::0/64"}
+				err                          error
+			)
+
+			BeforeAll(func() {
+				By("Setting test iteration parameters")
+				_, _, _, _, addressPool, _, err =
+					metallbenv.DefineIterationParams(
+						ipv4metalLbIPList, ipv6metalLbIPList, ipv4NodeAddrList, ipv6NodeAddrList, netparam.IPV4Family)
+				Expect(err).ToNot(HaveOccurred(), "Fail to set iteration parameters")
+			})
+
+			AfterEach(func() {
+				By("Clean metallb operator and test namespaces")
+				resetOperatorAndTestNS()
```

Comment:
Please move all funcs that you reuse to the common or to the internall folder

Reply:
Im not sure we will use any of these functions outside this test case file.

Reply:
They are currently in the internal>frr folder. Do you prefer them in the common?